Returning an array of array (using array[] notation) fails

Seldaek opened the pull request:
Accessing the variable that's been returned as `array[]` triggers `TypeError Suspicious array access to \Namespace\array` which I assume means it saw `array[]` as an array of objects of class `array`, but array is not a valid class name IIRC, and especially in type hints it's always used for actual arrays so I think it's safe to assume it's just equivalent to returning an array.. or array of arrays if you can express that.

Also the first commit removes two stray empty test files that got lost there it seems.

staabm commented:
Your rename also changed mode-bits to 755 (+x)

rlerdorf commented:
The namespace part doesn't seem like it is relevant here. You see the same from:
```
<?php
class ArrayProducer {
        /**
         * @return array[] array of arrays
         */
        public function getArrays()
        {
            return [['name' => 'lala'], ['name' => 'lulu']];
        }
}

$producer = new ArrayProducer;
$arrays = $producer->getArrays();
$test = $arrays[0]['name'];
```

Seldaek commented:
Indeed it isn't, I just started off with my situation but I didn't reduce it to the max.. anyway I updated the test (and fixed chmod.. not sure wtf happened there).
